src/assets/VueOptionVsCompositionAPI/option-vs-composition-api.md

```diff
@@ -0,0 +1,188 @@
+import { reactive, toRefs, computed , watch} from "vue";
+export default {
+  setup() {
+    // data
+    const data = reactive({
+      name: 'Sagar Gurung',
+      age: 26,
+      votingStatus: "Not Eligible",
+      bornYear: 1997
+    })
+    const mainData = toRefs(data)
+    // methods
+    function increaseAge() {
+      data.age++
+    }
+    function decreaseAge() {
+      data.age--
+    }
+    //computed
+    const getVotingStatus = computed(() => {
+      return (data.age >= 18) ? 'Eligible' : 'Not Eligible'
+    })
+    //watcher
+    watch(mainData.age, (oldAge, newAge) => {
+      if(oldAge > newAge) {
+        data.bornYear--
+      } else {
+        data.bornYear++
+      }
+    })
+
+    return {
+      ...mainData,
+      increaseAge,
+      decreaseAge,
+      getVotingStatus
+    }
+  }
+}
+```
+In the final code, we have imported `reactive`, `toRefs`, `computed`, and `watch` which are needed for building components with the `Composition API`. `toRefs` is used to convert the data created with `reactive` to plain objects. Also, when using `Composition API` we need to define all those `data`, `computed`, `watch`, and `methods` inside of function `setup()` as done above.
+At last, all the things that have been defined such as `data`, `computed`, and `methods` need to be returned to be exposed to the template.
+
+In this demo application we have learnt the basics of how to convert a component defined with `Option API` to `Composition API`. However, there are lots of other things that we can perform using the `Composition API`.
+I hope this blog gave you some insights on how to use the `Composition API` and also how you can replace your component defined in `Option API` with it.
+If you want to learn more about both `Options API` and `Composition API` then you can learn more from [here](https://vuejs.org/api/composition-api-setup.html).
+
+But before finishing the blog I would like to highlight some points regarding both `Options API` and `Composition API` so that you can relate which one to prefer before building up your projects.
+
+## `Options API` and `Composition API`
+
+You can find a lot of points regarding the comparison between these two APIs, both approaches for building up components have their benefits and drawbacks. I will only highlight five differences which I consider most relevant for choosing whether to use the Options API or Composition API in your own project:
+
+1. Composition API is new to Vue which can be called a modern way to define components whereas on the other hand Options API is proven or traditional way to build up components in Vue.
+2. Also, the structure and syntax for the Options API seems a bit simpler and easier to understand than that of the Composition API. It's a good choice for small projects and also for beginners. But for large and more complex projects, a better choice would be Composition API since it has a more flexible and modular structure that enhances better code maintainability.
```

Comment:
```suggestion
2. Also, the structure and syntax for the `Options API` seems a bit simpler and easier to understand than that of the `Composition API`. It's a good choice for small projects and also for beginners. But for large and more complex projects, a better choice would be `Composition API` since it has a more flexible and modular structure that enhances better code maintainability.
```